### PR TITLE
GPII-3682: Add Screen Capture button to the QS

### PR DESCRIPTION
**Pull request by Karadaliev**
* Added the new button in the settings and the message bundles
* Created widget based on the menuWidget
* Added all the required links between main and render in order the access to the siteConfig to work inside the widgets (shareXPath variable is used for the capture tool's path)

### CODE REVIEW

**Comment by Karadaliev** [comment]
@amb26 We will add the tests shortly, but you can start looking into it, and please be gentle 😺

**Comment by amb26** [comment]
ok to test

File: src/renderer/common/js/utils.js [hunk 1]
```diff
@@ -42,6 +44,24 @@
...
+     * @param command {String} - shareX command, example: "Morphic: Capture entire screen to desktop"
+     * @param shareXPath {String} - the path and executable name, example: "C:\\sharex-portable\\sharex.exe"
+     */
+    gpii.psp.execShareXCommand = function (command, shareXPath) {
+        // creates the command line, it should looks something like:
+        // "C:\\sharex-portable\\sharex.exe" -workflow "Morphic: Capture entire screen to desktop"
+        let commandToExecute = "\""+shareXPath+"\" -workflow \""+command+"\"";
+
+        try {
+            child_process.exec(commandToExecute);
```
Comment (amb26) [inline src/renderer/common/js/utils.js]: Should return something - e.g. a promise for the completion of the command

File: src/renderer/qssWidget/js/qssScreenCaptureWidget.js [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+"use strict";
+(function (fluid) {
+    var gpii = fluid.registerNamespace("gpii");
+
+    /**
+     * Represents the QSS Screen Capture widget.
+     */
+    fluid.defaults("gpii.qssWidget.screenCapture", {
+        gradeNames: ["fluid.viewComponent", "gpii.psp.heightObservable", "gpii.psp.selectorsTextRenderer"],
+        model: {
```
Comment (amb26) [inline src/renderer/qssWidget/js/qssScreenCaptureWidget.js]: Document in your members that there is a volatile member used for the keyboard event - I recommend something like
````
members: {
    // Holds the last keyboard event received during the "close" operation
    closingKeyboardEvent: null
    }
````

File: src/renderer/qssWidget/js/qssScreenCaptureWidget.js [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+                    }
+                }
+            }
+        },
+        listeners: {
+            "onCreate.visibilityChange": {
+                funcName: "gpii.qssWidget.screenCapture.addVisibilityChangeListener",
+                args: ["{closeTimer}"]
+            },
+            "onDestroy.visibilitychange": {
```
Comment (amb26) [inline src/renderer/qssWidget/js/qssScreenCaptureWidget.js]: capitalisation visibilityChange to match the other

File: src/renderer/common/js/utils.js (hunk 1 above)
Comment (Karadaliev) [reply src/renderer/common/js/utils.js]: Done

File: src/renderer/qssWidget/js/qssScreenCaptureWidget.js (hunk 2 above)
Comment (Karadaliev) [reply src/renderer/qssWidget/js/qssScreenCaptureWidget.js]: Done

**Comment by amb26** [comment]
Exit status 3221225725 apparently is "stack overflow", seemingly received during mouse wheel tests

**Comment by amb26** [comment]
ok to test

**Comment by amb26** [comment]
This one is now also in conflict

**Comment by Karadaliev** [comment]
@amb26 seems like everything its alright now, can we proceed with the merge?

**Comment by amb26** [comment]
Not quite yet - big conflicts again :)

**Comment by amb26** [comment]
Same failure as the first one in https://github.com/GPII/gpii-app/pull/99
````
01:54:39.367:  Data loaded for database 'gpii'...
01:54:43.322:  Test case listener has not responded after 5000ms - at sequence pos 3 of 12 sequence element {
    "event": "{testEnvironment}.events.noUserLoggedIn",
    "listener": "fluid.identity"
} of fixture GPII application integration tests
````

**Comment by amb26** [comment]
ok to test

**Comment by amb26** [comment]
Still fails a lot of linting

**Comment by amb26** [comment]
ok to test

**Comment by amb26** [comment]
and once for luck - ok to test

**Comment by amb26** [comment]
Got a local test case hang:
````
16:21:08.770:  Test case listener has not responded after 5000ms - at sequence pos 111 of 176 sequence element {
    "changeEvent": "{that}.app.qssWrapper.qssWidget.applier.modelChanged",
    "path": "isShown",
    "listener": "jqUnit.assertTrue",
    "args": [
        "The QSS widget is shown when the QSS button is activated using spacebar",
        "{that}.app.qssWrapper.qssWidget.model.isShown"
    ]
} of fixture QSS Widget integration tests
````

**Pull request merged**